packages/api/src/routers/tab.ts

```diff
@@ -63,129 +70,89 @@ export const tabRouter = createTRPCRouter({
-              creditorID: input.debtorID,
-            },
-            data: {
-              amount: inverseAmount,
-            },
-          });
-          return createdTab.amount;
-        }
-      } else {
-        if (!inverseAmount) {
-          const createdTab = await ctx.prisma.tab.updateMany({
-            where: {
-              debtorID: input.debtorID,
-              creditorID: input.creditorID,
+              debtorAndCreditor: {
+                debtorID: input.creditorID,
+                creditorID: input.debtorID,
+              },
             },
             data: {
               amount: {
-                increment: input.amount,
+                decrement: input.amount,
               },
             },
           });
-
-          const updatedTab = await ctx.prisma.tab.findFirst({
-            where: {
-              debtorID: input.debtorID,
-              creditorID: input.creditorID,
-            },
-          });
-          return updatedTab?.amount;
+          return inverseTab;
         } else {
-          if (tab.amount + input.amount > inverseAmount) {
-            tabAmount = tab.amount + input.amount - inverseAmount;
-            inverseAmount = 0;
-          } else if (tab.amount + input.amount < inverseAmount) {
-            tabAmount = 0;
-            inverseAmount = inverseAmount - (tab.amount + input.amount);
-          } else {
-            tabAmount = 0;
-            inverseAmount = 0;
-          }
-
-          const updatedTab = await ctx.prisma.tab.updateMany({
+          const newInverseTab = await ctx.prisma.tab.update({
             where: {
-              debtorID: input.debtorID,
-              creditorID: input.creditorID,
+              debtorAndCreditor: {
+                debtorID: input.creditorID,
+                creditorID: input.debtorID,
+              },
             },
             data: {
-              amount: tabAmount,
+              amount: 0,
             },
           });
 
-          const updatedInverseTab = await ctx.prisma.tab.updateMany({
+          const newTab = await ctx.prisma.tab.upsert({
             where: {
-              debtorID: input.creditorID,
-              creditorID: input.debtorID,
-            },
-            data: {
-              amount: inverseAmount,
+              debtorAndCreditor: {
+                debtorID: input.debtorID,
+                creditorID: input.creditorID,
+              },
             },
-          });
-
-          const updatedTabFind = await ctx.prisma.tab.findFirst({
-            where: {
+            create: {
               debtorID: input.debtorID,
               creditorID: input.creditorID,
+              amount: input.amount,
+            },
+            update: {
+              amount: {
+                increment: input.amount,
+              },
             },
           });
 
-          return updatedTabFind?.amount;
+          return newTab;
         }
       }
     }),
+
   clear: publicProcedure
```

Comment:
Clear does not clear A & B - it clears the tab A->B

meaning if tab is B->A then it will still exist;

confusion behaviour - maybe change?